src/app/routes/connector-ui/contract-agreement-page/contract-agreement-page/contract-agreement-page.service.ts

```diff
@@ -105,4 +123,23 @@ export class ContractAgreementPageService {
       filter((it) => it.isReady),
     );
   }
+
+  private fetchLimits(): Observable<ConnectorLimits | null> {
+    if (this.activeFeatureSet.hasConnectorLimits()) {
+      return this.edcApiService.getEnterpriseEditionConnectorLimits();
+    }
+    return of(null);
+  }
+
+  private setStatus(maxConnectorLimit: number | null) {
+    this.consumingContractAgreements.map((it, index) => {
```

Comment:
this is also a wrong use of map.

Map maps each array value to a new element, returning a new array.

In this case, you mutate each existing element, so if it was fine like this, forEach would have been more correct.

In general it's best practice to not mutate objects, but always create new objects.

For example, in this case:

```ts
return agreements.map((it, index) => ({
  ...it, // copy all existing fields

  // override fields where necessary
  statusText: 'something'
});
```

and you'll have a new array with new objects.

mutating is bad in ui development in general, because the change detection engines struggle if the object identities remain the same, but some deeply nested value changed.